internal/cmd/perfcomp/main.go

```diff
@@ -0,0 +1,287 @@
+		Order        int64  `bson:"order"`
+		TaskName     string `bson:"task_name"`
+		TaskID       string `bson:"task_id"`
+		Execution    int64  `bson:"execution"`
+		Mainline     bool   `bson:"mainline"`
+		OverrideInfo struct {
+			OverrideMainline bool        `bson:"override_mainline"`
+			BaseOrder        interface{} `bson:"base_order"`
+			Reason           interface{} `bson:"reason"`
+			User             interface{} `bson:"user"`
+		}
+		TestName string                 `bson:"test_name"`
+		Args     map[string]interface{} `bson:"args"`
+	}
+	CreatedAt   interface{} `bson:"created_at"`
+	CompletedAt interface{} `bson:"completed_at"`
+	Rollups     struct {
+		Stats []struct {
+			Name     string      `bson:"name"`
+			Val      float64     `bson:"val"`
+			Metadata interface{} `bson:"metadata"`
+		}
+	}
+	FailedRollupAttempts int64 `bson:"failed_rollup_attempts"`
+}
+
+type StableRegion struct {
+	TimeSeriesInfo struct {
+		Project     string                 `bson:"project"`
+		Variant     string                 `bson:"variant"`
+		Task        string                 `bson:"task"`
+		Test        string                 `bson:"test"`
+		Measurement string                 `bson:"measurement"`
+		Args        map[string]interface{} `bson:"args"`
+	}
+	Start                  interface{}   `bson:"start"`
+	End                    interface{}   `bson:"end"`
+	Values                 []float64     `bson:"values"`
+	StartOrder             int64         `bson:"start_order"`
+	EndOrder               int64         `bson:"end_order"`
+	Mean                   float64       `bson:"mean"`
+	Std                    float64       `bson:"std"`
+	Median                 float64       `bson:"median"`
+	Max                    float64       `bson:"max"`
+	Min                    float64       `bson:"min"`
+	CoefficientOfVariation float64       `bson:"coefficient_of_variation"`
+	LastSuccessfulUpdate   interface{}   `bson:"last_successful_update"`
+	Last                   bool          `bson:"last"`
+	Contexts               []interface{} `bson:"contexts"`
+}
+
+type EnergyStats struct {
+	Benchmark    string
+	Measurement  string
+	PatchVersion string
+	StableRegion StableRegion
+	PatchValues  []float64
+	PChange      float64
+	E            float64
+	T            float64
+	H            float64
+	Z            float64
+}
+
+func main() {
+	// Connect to analytics node
+	uri := os.Getenv("PERF_URI_PRIVATE_ENDPOINT")
+	if uri == "" {
+		log.Panic("PERF_URI_PRIVATE_ENDPOINT env variable is not set")
+	}
+
+	version := os.Getenv("VERSION_ID")
+	if version == "" {
+		log.Panic("VERSION_ID env variable is not set")
+	}
+
+	client, err1 := mongo.Connect(options.Client().ApplyURI(uri))
+	if err1 != nil {
+		log.Panicf("Error connecting client: %v", err1)
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+	err2 := client.Ping(ctx, nil)
+	if err2 != nil {
+		log.Panicf("Error pinging MongoDB Analytics: %v", err2)
```

Comment:
Suggest using `log.Fatalf`, no need to panic here. This applies to all other instances of log.Panicf.